File: gcp-pubsub/send-server/model/email.go
package model

import (
	"gorm.io/gorm"
)

type EmailStatus string

const (
	Pending EmailStatus = "PENDING"
	Sent    EmailStatus = "SENT"
	FAILED  EmailStatus = "FAILED"
)

type Email struct {
	// Base Column
	gorm.Model

	Title    string      `gorm:"column:title;not null"`
	Body     string      `gorm:"column:title;not null"`
	Receiver string      `gorm:"column:receiver;not null"`
	Status   EmailStatus `gorm:"column:status;not null"`
}

// 테이블명 매핑
func (Email) TableName() string {
	return "email"
}

File: gcp-pubsub/api-server/src/main/java/com/springboot/pubsub/domain/Email.java
package com.springboot.pubsub.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "email")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Email {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "body", nullable = false)
    private String body;

    @Column(name = "receiver", nullable = false)
    private String receiver;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private EmailStatus status;

    @Builder
    public Email(String title, String body, String receiver) {
        this.title = title;
        this.body = body;
        this.receiver = receiver;
        this.status = EmailStatus.PENDING;
    }
}
